### PR TITLE
Use `# optional - gap_package_grape` etc. instead of `# optional - gap_packages`

### PR DESCRIPTION
**Pull request by mkoeppe**
The Sage distribution has an optional SPKG `gap_packages`, which is a collection of GAP packages used by parts of Sage. 
The doctests that uses these packages are currently marked `# optional - gap_packages`. Here we switch to using more specific feature tags. These more specific features are tested dynamically, which works outside of the Sage distribution.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

- Depends on #36017

### CODE REVIEW

File: src/sage/groups/perm_gps/permgroup.py [hunk 1]
```diff
@@ -182,13 +182,13 @@ def hap_decorator(f):
         sage: from sage.groups.perm_gps.permgroup import hap_decorator
         sage: def foo(self, n, p=0): print("Done")
         sage: foo = hap_decorator(foo)
-        sage: foo(None, 3)    #optional - gap_packages
+        sage: foo(None, 3)    #optional - gap_package_hap
```
Comment (dcoudert) [inline src/sage/groups/perm_gps/permgroup.py]: `#optional` -> `# optional`, and we could use a block stop here

File: src/sage/groups/perm_gps/permgroup.py [hunk 2]
```diff
@@ -182,13 +182,13 @@ def hap_decorator(f):
...
         Done
-        sage: foo(None, 3, 0) # optional - gap_packages
+        sage: foo(None, 3, 0) # optional - gap_package_hap
         Done
-        sage: foo(None, 3, 5) # optional - gap_packages
+        sage: foo(None, 3, 5) # optional - gap_package_hap
         Done
-        sage: foo(None, 3, 4) #optional - gap_packages
+        sage: foo(None, 3, 4) #optional - gap_package_hap
```
Comment (dcoudert) [inline src/sage/groups/perm_gps/permgroup.py]: `#optional` -> `# optional`

File: src/sage/groups/perm_gps/permgroup_named.py [hunk 3]
```diff
@@ -1063,14 +1063,14 @@ def __init__(self, n):
 
         EXAMPLES::
 
-            sage: G = groups.permutation.Janko(1); G # optional - gap_packages internet
+            sage: G = groups.permutation.Janko(1); G # optional - gap_package_atlasrep internet
```
Comment (dcoudert) [inline src/sage/groups/perm_gps/permgroup_named.py]: double space before `# optional`, many places

**Review by dcoudert — CHANGES_REQUESTED** [review]
why do we have so many different packages for gap and how should one know which one to use ?

**Comment by dimpase** [comment]
On 5 August 2023 12:35:41 BST, David Coudert ***@***.***> wrote:
>@dcoudert requested changes on this pull request.
>
>why do we have so many different packages for gap and how should one know which one to use ?

it is usually clear from GAP's LoadPackage calls in Sage code.

>
>> @@ -182,13 +182,13 @@ def hap_decorator(f):
>         sage: from sage.groups.perm_gps.permgroup import hap_decorator
>         sage: def foo(self, n, p=0): print("Done")
>         sage: foo = hap_decorator(foo)
>-        sage: foo(None, 3)    #optional - gap_packages
>+        sage: foo(None, 3)    #optional - gap_package_hap
>
>`#optional` -> `# optional`, and we could use a block stop here
>
>>          Done
>-        sage: foo(None, 3, 4) #optional - gap_packages
>+        sage: foo(None, 3, 4) #optional - gap_package_hap
>
>`#optional` -> `# optional`
>
>> @@ -1063,14 +1063,14 @@ def __init__(self, n):
> 
>         EXAMPLES::
> 
>-            sage: G = groups.permutation.Janko(1); G # optional - gap_packages internet
>+            sage: G = groups.permutation.Janko(1); G # optional - gap_package_atlasrep internet
>
>double space before `# optional`, many places
>

**Comment by mkoeppe** [comment]
At least in some cases, also the docstrings contained information about which GAP package is needed.

**Review by dcoudert — APPROVED** [review]
I heve no more comments. LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**